### PR TITLE
Prevent SFDMP from throwing internal errors

### PR DESCRIPTION
**Pull request by brennan**
**What does this PR do?**
This PR prevents SFDMP from throwing errors when an unmapped `track` event is invoked while the `trackEventNames` setting is enabled.

**Are there breaking changes in this PR?**
n/a

**Has this been tested end-to-end? Please provide screenshots on how the fix now populates in the end tool. If not, what was done to test?**
n/a

**Any background context you want to provide?**
Should reduce `track`-related errors to 0 - https://segment.datadoghq.com/dash/807059/analyticsjs-integration-dashboard?live=true&page=0&is_auto=false&from_ts=1539186716844&to_ts=1539201116844&tile_size=m&tpl_var_integration_name=salesforce_dmp

**Is there parity with the server-side/android/iOS integration (if applicable)?**
n/a

**Does this require a metadata change? If so, please link the PR from https://github.com/segmentio/destination-scripts.**
n/a

**What are the relevant tickets?**
n/a

**Link to CC ticket**

**List all the tests accounts you have used to make sure this change works**
n/a

**Helpful Docs**
n/a

### CODE REVIEW

**Comment by calvinfo** [comment]
This looks reasonable, can we add test coverage to catch this case? There should be some accompanying test here which fails without the modification

**Comment by brennan** [comment]
@calvinfo Will do!

**Comment by brennan** [comment]
@calvinfo Having trouble using our test suite to validate that a method doesn't throw an error. We can come back and add a test for this resolved edge case later; otherwise, we may not be able to deploy the fix today. Let me know what you think!

File: integrations/salesforce-dmp/test/index.test.js [hunk 1]
```diff
@@ -188,6 +189,27 @@ describe('Salesforce DMP V2', function () {
         analytics.track('event')
         analytics.assert(window.kruxDataLayer['event_name'] === 'do not override')
       })
+
+      // previously SFDMP would throw an error if a customer invoked an 
+      // unmapped `track` event while `sendEventNames` setting was enabled
+      // this edge case has been resolved, but we test for it anyway
+      it.only('throws an error if `sendEventNames` setting is enabled but no track events are whitelisted', function () {
```
Comment (ucarion) [inline integrations/salesforce-dmp/test/index.test.js]: I think the `only` thing is accidentally left over?

File: integrations/salesforce-dmp/lib/index.js [hunk 2]
```diff
@@ -117,6 +117,8 @@ SalesforceDMP.prototype.identifyV2 = SalesforceDMP.prototype.trackV2 = function
         }
         if (i === this.options.trackFireEvents.length - 1) return
       }
+    } else {
+      return
```
Comment (ucarion) [inline integrations/salesforce-dmp/lib/index.js]: This seems like a pretty big change? Do we have tests for this early return?

File: integrations/salesforce-dmp/lib/index.js [hunk 3]
```diff
@@ -137,22 +139,24 @@ SalesforceDMP.prototype.identifyV2 = SalesforceDMP.prototype.trackV2 = function
...
   }
 
-  if (this.options.sendEventNames && !window.kruxDataLayer['event_name']) {
-    var name
-    if (type === 'track') {
-      name = msg.event()
-    } else {
-      if (this.options.sendIdentifyAsEventName) {
-        name = type
+  if (window.kruxDataLayer) {
```
Comment (ucarion) [inline integrations/salesforce-dmp/lib/index.js]: The diff GitHub is showing here is just from indenting the code because of this `if`, right?

File: integrations/salesforce-dmp/test/index.test.js [hunk 4]
```diff
@@ -188,6 +189,27 @@ describe('Salesforce DMP V2', function () {
...
+        var edgeCaseOptions = {
+          confId: 'rw1to29bb',
```
Comment (ucarion) [inline integrations/salesforce-dmp/test/index.test.js]: Is this value sensitive?

File: integrations/salesforce-dmp/lib/index.js (hunk 2 above)
Comment (ucarion) [reply integrations/salesforce-dmp/lib/index.js]: Hmm, actually it seems like this early return is the meat of your fix. Maybe we could put a comment explaining why it needs to be there / what the consequences of leaving it out are?
